src/site/content/en/blog/optimize-lcp/index.md

```diff
@@ -0,0 +1,644 @@
+frameworks need to create the web page dynamically. In other words, rather than just sending a
+complete HTML file that's already ready when the browser requests it, the frameworks need to run
+logic to construct the page. This could be due to pending results from a database query or even
+because components need to be generated into markup by a UI framework (such as
+[React](https://reactjs.org/docs/react-dom-server.html)). Many web frameworks that run on the server
+have performance guidance that you can use to speed up this process.
+
+{% Aside %}
+Check out [Fix an overloaded server](/overloaded-server/) for more tips.
+{% endAside %}
+
+### Route users to a nearby CDN
+
+A Content Delivery Network (CDN) is a network of servers distributed in many different locations. If
+the content on your web page is being hosted on a single server, your website will load slower for
+users that are geographically further away because their browser requests literally have to travel
+around the world. Consider using a CDN to ensure that your users never have to wait for network
+requests to faraway servers.
+
+### Cache assets
+
+If your HTML is static and doesn't need to change on every request, caching can prevent it from
+being recreated unnecessarily. By storing a copy of the generated HTML on disk, server-side caching
+can reduce TTFB and minimize resource usage.
+
+Depending on your toolchain, there are many different ways to apply server caching:
+
++   Configure reverse proxies ([Varnish](https://varnish-cache.org/),
+    [nginx](https://www.nginx.com/)) to serve cached content or act as a cache server when
+    installed in front of an application server
++   Configure and manage your cloud provider's
+    ([Firebase](https://firebase.google.com/docs/hosting/manage-cache),
+    [AWS](https://aws.amazon.com/caching/),
+    [Azure](https://docs.microsoft.com/en-us/azure/architecture/best-practices/caching)) cache behavior
++   Use a CDN that provides edge servers so that your content is cached and stored closer to
+    your users
+
+### Serve HTML pages cache-first
+
+When installed, a [service
+worker](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) runs in the browser
+background and can intercept requests from the server. This makes it possible to cache the HTML page
+and only go to the network when the file's content has changed.
+
+The following chart shows how LCP distributions have been reduced on a site using this pattern: 
+
+<figure class="w-figure">
+  <img src="./lcp-sw-caching.png" alt="Largest Contentful Paint distributions before and after HTML caching">
+  <figcaption class="w-figcaption">Largest Contentful Paint distribution, with and without a service worker - <a href="https://philipwalton.com/articles/smaller-html-payloads-with-service-workers/">philipwalton.com</a></figcaption>
+</figure>
+
+This chart shows the number of page loads on a single site within the past 30 days with varying LCP
```

Comment:
Suggest rewording: "The chart shows the distribution for LCP from a single site over the last 28 days, segmented by service worker state. Notice how far more page loads have a faster LCP value after cache-first HTML page serving was introduced in the service worker (blue portion of chart)."